angularSTEP/src/app/discover-display/discover-display.component.ts

```diff
@@ -0,0 +1,207 @@
+            .get()
+            .then(recipes => {
+              this.recipes = recipes.docs;
+            });
+        }
+        break;
+      }
+      case 'Name': {
+        if (this.direction) {
+          this.recipeCollection.ref
+            .orderBy('recipeName', 'desc')
+            .withConverter(new RecipeConverter().recipeConverter)
+            .get()
+            .then(recipes => {
+              this.recipes = recipes.docs;
+            });
+        } else {
+          this.recipeCollection.ref
+            .orderBy('recipeName', 'asc')
+            .withConverter(new RecipeConverter().recipeConverter)
+            .get()
+            .then(recipes => {
+              this.recipes = recipes.docs;
+            });
+        }
+        break;
+      }
+    }
+  }
+
+  getUsers() {
+    switch (this.sortType) {
+      case 'Time Created': {
+        if (this.direction) {
+          this.userCollection.ref
+            .orderBy('time', 'asc')
+            .withConverter(new Converter().userConverter)
+            .get()
+            .then(users => {
+              this.users = users.docs;
+            });
+        } else {
+          this.userCollection.ref
+            .orderBy('time', 'desc')
+            .withConverter(new Converter().userConverter)
+            .get()
+            .then(users => {
+              this.users = users.docs;
+            });
+        }
+        break;
+      }
+      case 'Number of Recipes': {
+        //temporary until num of recipes are actually added
+        if (this.direction) {
+          this.userCollection.ref
+            .orderBy('displayName', 'desc')
+            .withConverter(new Converter().userConverter)
+            .get()
+            .then(users => {
+              this.users = users.docs;
+            });
+        } else {
+          this.userCollection.ref
+            .orderBy('displayName', 'asc')
+            .withConverter(new Converter().userConverter).limit(10)
+            .get()
+            .then(users => {
+              this.users = users.docs;
+            });
+        }
+        break;
+      }
+      case 'Name': {
+        if (this.direction) {
+          this.userCollection.ref
+            .orderBy('displayName', 'desc')
+            .withConverter(new Converter().userConverter).limit(10)
+            .get()
+            .then(users => {
+              this.users = users.docs;
+            });
+        } else {
+          this.userCollection.ref
+            .orderBy('displayName', 'asc')
+            .withConverter(new Converter().userConverter).limit(10)
+            .get()
+            .then(users => {
+              this.users = users.docs;
+            });
+        }
+        break;
+      }
+    }
+  }
+
+  goToUser(username: string) {
+    this.router.navigate(['discover/users/' + username]);
+  }
+
+  goToRecipe(id: string) {
+    if(id){
+      this.router.navigate(['discover/recipes', id]);
```

Comment:
Should the forward slash be in `this.router.navigate(['discover/users/' + username]);` or is `this.router.navigate(['discover/recipes', id]);` missing the forward slash

Reply:
Oh yeah, thanks. I just changed them so they are uniform.